File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.java
import java.util.*;

class Solution
{
    public int solution(String s)
    {
        Stack<Character> stack = new Stack<>();
        char[] chars = s.toCharArray();
        for(char c : chars) {
            if(stack.isEmpty()) stack.push(c);
            else {
                if(stack.peek() == c) stack.pop();
                else stack.push(c);
            }
        }
        return stack.isEmpty() ? 1 : 0;
    }
}

File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.swift
import Foundation

func solution(_ s:String) -> Int{
    var stack:[String] = []
    var len = 0
    let list = s.map{String($0)}
    for i in 0..<list.count {
        if(stack.isEmpty) { 
            stack.append(list[i])
            len += 1
        }
        else {
            if(stack[len-1] == list[i]) { 
                stack.removeLast()
                len -= 1
            }
            else { 
                stack.append(list[i])
                len += 1
            }
        }
    }
    return stack.isEmpty ? 1 : 0
}